docs/api/permissions.rst

```diff
@@ -0,0 +1,332 @@
+| Record         | ``write``              | Ability to write and read the    |
+|                |                        | record.                          |
+|                +------------------------+----------------------------------+
+|                | ``read``               | Ability to read the record.      |
+|                |                        |                                  |
++----------------+------------------------+----------------------------------+
+| Group          | ``write``              | Ability to write and read the    |
+|                |                        | group.                           |
+|                +------------------------+----------------------------------+
+|                | ``read``               | Ability to read the group.       |
+|                |                        |                                  |
++----------------+------------------------+----------------------------------+
+
+
+By default the ``write`` permission is given to the creator of an
+object.
+
+
+.. note::
+
+  There is no ``delete`` permission: Anyone with the ``write`` permission on an
+  object can also edit its associated permissions and delete it.
+
+
+Principals
+==========
+
+During the authentication phase, a set of :term:`principal`s for the current
+authenticated *user* will be bound to to the request.
+
+The main principal is considered the **user id**, and follows this formalism:
+``{type}:{identifier}`` (e.g. for Firefox Account: ``fxa:32aa95a474c984d41d395e2d0b614aa2``).
+
+There are two special principals:
+
+- ``system.Authenticated``: All users that are authenticated, no matter the
+  authentication mean.
+- ``system.Everyone``: Anyone (authenticated or anonymous). Using this
+  principal is useful when a rule should apply to all users.
+
+.. note::
+
+    A user can also be another application (in order to provide *service to
+    service* authentication).
+
+Get the current user id
+-----------------------
+
+The currently authenticated *user id* can be obtained on the root url.
+
+.. code-block:: http
+    :emphasize-lines: 16
+
+    $ http GET http://localhost:8888/v1/ --auth user:pass
+    HTTP/1.1 200 OK
+    Access-Control-Expose-Headers: Backoff, Retry-After, Alert, Content-Length
+    Content-Length: 288
+    Content-Type: application/json; charset=UTF-8
+    Date: Thu, 16 Jul 2015 09:48:47 GMT
+    Server: waitress
+
+    {
+        "documentation": "https://kinto.readthedocs.org/",
+        "hello": "cloud storage",
+        "settings": {
+            "cliquet.batch_max_requests": 25
+        },
+        "url": "http://localhost:8888/v1/",
+        "userid": "basicauth:631c2d625ee5726172cf67c6750de10a3e1a04bcd603bc9ad6d6b196fa8257a6",
+        "version": "1.4.0"
+    }
+
+
+In this case the user ID is: ``basicauth:631c2d625ee5726172cf67c6750de10a3e1a04bcd603bc9ad6d6b196fa8257a6``
+
+.. note::
+
+    In case of sharing, users need a way to share their user ID with
```

Comment:
This is awkward. Consider:
> Objects are shared based on user ID.

That may be too brief. Two (more verbose) possibilities:
> By granting permissions on an object to another user ID, you can *share* that object with another user.
> You can *share* an object with another user by granting permissions on that object to another user ID.